### PR TITLE
Basic routes, logic, and design

### PR DESCRIPTION
**Pull request by rexolion**
## Related issue(s)

Resolves: None

#### Related changes (conditional)

- Tests
  - [ ] If I added new functionality, I added tests covering it.
  - [ ] If I fixed a bug, I added a regression test to prevent the bug from silently reappearing again.

- Documentation
  - [x] I checked whether I should update the docs and did so if necessary: [README](../tree/master/README.md), Haddock
  - [x] I added an entry to the [changelog](../tree/master/CHANGES.md) if my changes are visible to the users.

#### Before merging (mandatory)

- [x] My commits comply with [the policy used in Union](https://github.com/union-platform/guides/blob/master/commits.md) and code complies with the [styles guide](https://github.com/union-platform/guides/blob/master/README.md#programming-languages).
- [x] Check __target__  branch of PR is set correctly

### CODE REVIEW

**Comment by worm2fed** [comment]
great job! A lot of review for me 😅😭

**Comment by rexolion** [comment]
@worm2fed brief demo of what has been done (spoiler: a lot!)

Tests will be in other commits.

https://youtu.be/TBVb9FdcMXE

File: lib/mocks/interests_mock.dart [hunk 1]
```diff
@@ -0,0 +1,100 @@
+// SPDX-FileCopyrightText: © 2023 Union Platform <https://unionapp.cc>
+//
+// SPDX-License-Identifier: LicenseRef-Union
+
+import 'package:union_app/models/user_profile_model.dart';
+
+const InterestsMock = [
```
Comment (worm2fed) [inline lib/mocks/interests_mock.dart:7]: it's in DB, there is endpoint to get it or create a new one if it is not in list
but we can keep your list and use as fixture at backend side

File: lib/models/user_profile_model.dart [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+    this.motivation,
+    this.name,
+    this.phone,
+    this.telegram,
+  });
+
+  UserProfileModel.fromJson(Map<String, dynamic> json) {
+    accountId = json['accountId'] != null ? json['accountId'] as String : null;
+    activatedAt =
+        json['activatedAt'] != null ? json['activatedAt'] as String : null;
```
Comment (worm2fed) [inline lib/models/user_profile_model.dart:32]: uf.. is there a shortcut for this?

File: lib/models/user_profile_model.dart [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+    return data;
+  }
+}
+
+class Interests {
+  Interests({this.name, this.pk});
+
+  Interests.fromJson(Map<String, dynamic> json) {
+    name = json['name'] != null ? json['name'] as String : null;
+    pk = json['pk'] != null ? json['pk'] as int : null;
```
Comment (worm2fed) [inline lib/models/user_profile_model.dart:114]: both cannot be null

File: lib/models/user_profile_model.dart [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+    this.linkedIn,
+    this.mission,
+    this.motivation,
+    this.name,
+    this.phone,
+    this.telegram,
+  });
+
+  UserProfileModel.fromJson(Map<String, dynamic> json) {
+    accountId = json['accountId'] != null ? json['accountId'] as String : null;
```
Comment (worm2fed) [inline lib/models/user_profile_model.dart:30]: it cannot be null

File: lib/models/user_profile_model.dart [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+      for (final v in json['interests'] as List) {
+        interests!.add(Interests.fromJson(v as Map<String, dynamic>));
+      }
+    }
+    lifeSense = json['lifeSense'] != null ? json['lifeSense'] as String : null;
+    linkedIn = json['linkedIn'] != null ? json['linkedIn'] as String : null;
+    mission = json['mission'] != null ? json['mission'] as String : null;
+    motivation =
+        json['motivation'] != null ? json['motivation'] as String : null;
+    name = json['name'] != null ? json['name'] as String : null;
```
Comment (worm2fed) [inline lib/models/user_profile_model.dart:55]: it cannot be null

File: lib/screens/activity/create_team.dart [hunk 6]
```diff
@@ -0,0 +1,113 @@
+// SPDX-FileCopyrightText: © 2023 Union Platform <https://unionapp.cc>
+//
+// SPDX-License-Identifier: LicenseRef-Union
+
+import 'package:flutter/material.dart';
+
+var _currencies = [
```
Comment (worm2fed) [inline lib/screens/activity/create_team.dart:7]: currencies?))
for scope there is endpoint in DB (without creation one, I have fixture)

File: lib/screens/login.dart [hunk 7]
```diff
@@ -0,0 +1,137 @@
...
+      body: Center(
+        child: Padding(
+          padding: const EdgeInsets.only(left: 16, right: 16),
+          child: Column(
+            mainAxisAlignment: MainAxisAlignment.center,
+            children: [
+              SizedBox(
+                width: double.infinity,
+                child: Text(
+                  phoneSubmitted
```
Comment (worm2fed) [inline lib/screens/login.dart:81]: we switched to email and OTP from it

File: lib/widgets/project_card.dart [hunk 8]
```diff
@@ -0,0 +1,77 @@
...
+                        ),
+                      )
+                  ],
+                ),
+              ),
+            ],
+          ),
+        ),
+      ),
+    );
```
Comment (worm2fed) [inline lib/widgets/project_card.dart:75]: a lot of brackets))

**Review by worm2fed — CHANGES_REQUESTED** [review]
wow, good job

**Comment by worm2fed** [comment]
@rexolion pls, rename your first commit as described here: https://www.conventionalcommits.org/en/v1.0.0/

File: .gitignore [hunk 9]
```diff
@@ -1,13 +1,44 @@
-# SPDX-FileCopyrightText: © 2021-2023 Union Platform <https://unionapp.cc>
-# SPDX-License-Identifier: CC0-1.0
+# Miscellaneous
+*.class
+*.log
+*.pyc
+*.swp
+.DS_Store
```
[the remark this answers is not in the document]
Comment (rexolion) [reply .gitignore:6]: What if someone fucked up that one? I want a fool proof system

File: android/app/build.gradle [hunk 10]
```diff
@@ -0,0 +1,71 @@
...
+}
+
+def flutterVersionCode = localProperties.getProperty('flutter.versionCode')
+if (flutterVersionCode == null) {
+    flutterVersionCode = '1'
+}
+
+def flutterVersionName = localProperties.getProperty('flutter.versionName')
+if (flutterVersionName == null) {
+    flutterVersionName = '1.0'
```
[the remark this answers is not in the document]
Comment (rexolion) [reply android/app/build.gradle:21]: It in pubspec for android (and inner flutter) and in Runner for ios

[the remark this answers is not in the document]
Comment (rexolion) [reply android/app/build.gradle:21]: I set the version you like

File: android/app/src/main/kotlin/com/example/app/MainActivity.kt [hunk 11]
```diff
@@ -0,0 +1,6 @@
+package com.example.app
```
[the remark this answers is not in the document]
Comment (rexolion) [reply android/app/src/main/kotlin/com/example/app/MainActivity.kt:1]: 'Truly' renaming the app it's a task for a different commit

File: android/app/src/main/res/mipmap-hdpi/ic_launcher.png
[the remark this answers is not in the document]
Comment (rexolion) [reply android/app/src/main/res/mipmap-hdpi/ic_launcher.png:1]: Branding the app it's a task for a different commit

File: lib/mocks/gallery_mock.dart
[the remark this answers is not in the document]
Comment (rexolion) [reply lib/mocks/gallery_mock.dart:1]: I don't get the question. See the demo, it's the gallery of photos in the project entity.

File: lib/mocks/interests_mock.dart (hunk 1 above)
Comment (rexolion) [reply lib/mocks/interests_mock.dart:7]: It's just the mock for the initial commit of routes, I need some list to use it in UI. Surely we will get it through the API in future commits

File: lib/models/user_profile_model.dart (hunk 2 above)
Comment (rexolion) [reply lib/models/user_profile_model.dart:32]: No, dart is strictly null safe

File: lib/models/user_profile_model.dart (hunk 3 above)
Comment (rexolion) [reply lib/models/user_profile_model.dart:114]: ok

File: lib/models/user_profile_model.dart (hunk 5 above)
Comment (rexolion) [reply lib/models/user_profile_model.dart:55]: mission and motivation could be null, name not

File: lib/screens/activity/create_team.dart (hunk 6 above)
Comment (rexolion) [reply lib/screens/activity/create_team.dart:7]: Renamed it, API integration in other commit

File: android/app/build.gradle (hunk 10 above)
[the remark this answers is not in the document]
Comment (worm2fed) [reply android/app/build.gradle:21]: Can we make one source of truth? To specify it in 1 place, not in 3

File: lib/models/user_profile_model.dart (hunk 5 above)
Comment (worm2fed) [reply lib/models/user_profile_model.dart:55]: Yes

File: android/app/build.gradle (hunk 10 above)
[the remark this answers is not in the document]
Comment (rexolion) [reply android/app/build.gradle:25]: Only in two, ios and android, as I know no, cannot specify it in one place.

[the remark this answers is not in the document]
Comment (rexolion) [reply android/app/build.gradle:25]: Here is constant used

**Comment by rexolion** [comment]
@worm2fed is it counts as BREAKING CHANGE?

File: lib/screens/login.dart (hunk 7 above)
Comment (rexolion) [reply lib/screens/login.dart:81]: Basic initial setup commit, I will fix it in other commit

**Comment by worm2fed** [comment]
@rexolion can I push to your PR?

**Comment by rexolion** [comment]
@worm2fed Ok, if needed :)

**Comment by rexolion** [comment]
@worm2fed I want to continue development in my spare time, so if it takes time for you to commit, let's do that in a different PR

**Comment by worm2fed** [comment]
@rexolion 10 min

File: android/app/src/debug/AndroidManifest.xml
Comment (worm2fed) [inline android/app/src/debug/AndroidManifest.xml:1]: should it be in ignore?

There is this in ignore`/android/app/debug`
but this is not: `/android/src/app/debug`

is it correct?

**Review by worm2fed — APPROVED** [review]

File: android/app/src/debug/AndroidManifest.xml
Comment (rexolion) [reply android/app/src/debug/AndroidManifest.xml:1]: yep

**Comment by rexolion** [comment]
@worm2fed You cannot add comments inside storyboard files — build fails, it's the guts of the ios, that's why I added the license in a different file.

image
image

**Comment by worm2fed** [comment]
@rexolion mm.. but build is done without errors

**Comment by rexolion** [comment]
> @rexolion mm.. but build is done without errors

It's Android's build in Github action

**Comment by worm2fed** [comment]
@rexolion will push fix now

**Comment by worm2fed** [comment]
@rexolion what about plist?

**Comment by rexolion** [comment]
> @rexolion what about plist?

Only storyboard files didn't like that

**Pull request merged**